iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_PUBLISHER_H_
+#define IOX_BINDING_C_PUBLISHER_H_
+
+#include "iceoryx_binding_c/internal/c2cpp_bridge.h"
+#include "iceoryx_binding_c/types.h"
+
+CLASS PublisherPortData*
+Publisher_new(const char* service, const char* instance, const char* event, const uint64_t historyCapacity);
```

Comment:
runnable name missing

Reply:
@budrus The idea was the following:
- Every publisher and subscriber must be created inside a runnable and then the runnable contains all ports.
- If you do not create a runnable the posh runtime comes with a default runnable in which those ports are created. This means the constructor is adding the port to the default runnable.
- The C API would then have two `Publisher_create` methods.
   1. The first one is the one above where the port is added to the default runnable
   2. The second `Pubisher_create` method would have another argument, the runnable handle, to which the port will be added.

The 2. part will be implemented in the follow up pull request.